### PR TITLE
[Uniform Schema] Use table name to load schema instead of schemaName field in tableConfig

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
This will enforce realtime and offline table use the same schema
Keep backward compatible but will log a warning when using deprecated way to load schema

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/metadata/ZKMetadataProvider.java [hunk 1]
```diff
@@ -208,39 +209,32 @@ public static Schema getSchema(@Nonnull ZkHelixPropertyStore<ZNRecord> propertyS
   }
 
   @Nullable
-  public static Schema getOfflineTableSchema(@Nonnull ZkHelixPropertyStore<ZNRecord> propertyStore,
+  public static Schema getTableSchema(@Nonnull ZkHelixPropertyStore<ZNRecord> propertyStore,
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/metadata/ZKMetadataProvider.java]: add a javadoc  coment that tableName arg can be with OR wthout table type.

File: pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/HelixInstanceDataManager.java [hunk 2]
```diff
@@ -231,17 +231,11 @@ public synchronized void removeSegment(String segmentName) {
...
       throws Exception {
     String segmentName = segmentMetadata.getName();
-    String tableName = segmentMetadata.getTableName();
-    if (tableType == CommonConstants.Helix.TableType.OFFLINE) {
-      tableName = TableNameBuilder.OFFLINE.tableNameWithType(tableName);
-    } else {
-      tableName = TableNameBuilder.REALTIME.tableNameWithType(tableName);
-    }
-    LOGGER.info("Reloading segment: {} in table: {}", segmentName, tableName);
+    LOGGER.info("Reloading segment: {} in table: {}", segmentName, tableNameWithType);
```
Comment (mcvsubbu) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/HelixInstanceDataManager.java]: please also add an info log when reload is successful with segment name and table name in the log.

File: pinot-common/src/main/java/com/linkedin/pinot/common/metadata/ZKMetadataProvider.java (hunk 1 above)
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/com/linkedin/pinot/common/metadata/ZKMetadataProvider.java]: Done.

File: pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/HelixInstanceDataManager.java (hunk 2 above)
Comment (Jackie-Jiang) [inline pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/HelixInstanceDataManager.java]: Added

**Pull request merged**